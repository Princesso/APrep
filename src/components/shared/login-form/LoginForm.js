import React from 'react'
import './login-form.css'
import '../../../styles/main.css'
import {Link, withRouter} from 'react-router-dom'
import {connect} from 'react-redux'
import {userLogin} from '../../../actions/authActions'

class LoginForm extends React.Component {
  state = {
    email: '',
    password:'',
  }

  handleChange = e => this.setState({ [e.target.name]: e.target.value });

  handleSubmit= async(e) => {
    e.preventDefault();
    const loginData = this.state;
    //write validation function
    this.setState({
      email: '',
      password:'',
    })
    this.props.userLogin(loginData)
  }

  componentDidMount() {
    if(this.props.auth.isAuthenticated === true)
    this.props.history.push('/dashboard') 
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.auth.isAuthenticated === true)
    this.props.history.push('/dashboard') 
  }
  render() {
    const {email, password} = this.state
    return(
      <div className="form-holder">
        <form className="form" method="post" onSubmit={this.handleSubmit}> 
          <h3>Login</h3>
          <div>
            <p>
              {/* render error */}
            </p>
          </div>
          <input 
            type="email" 
            placeholder="Email" 
            required 
            id="email"
            name="email"
            value={email}
            onChange={this.handleChange}
          />
          <input 
            type="password" 
            placeholder="Password" 
            required  
            id="password"
            name="password"
            value={password}
            onChange={this.handleChange}
          />
          <button type="submit" id="submitBtn">Login</button>
          <p id="message">Have no account? <Link to="/register" className="linker"> Register</Link></p>
        </form>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth
})

export default connect(mapStateToProps, {userLogin}) (withRouter(LoginForm))