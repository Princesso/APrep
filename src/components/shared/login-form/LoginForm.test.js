import React from 'react';
import {LoginForm} from './LoginForm';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Store from '../../../store';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

test('Login Form', () => {
  const props = {
    auth: {
      isAuthenticated: true,
      user: {}
    },
    location: { 
      pathname: '/', 
      search: '', 
      hash: '', 
      state: undefined 
    },
    computedMatch: { 
      path: '/', 
      url: '/', 
      params: {}, 
      isExact: true 
    },
    history: {
      push: jest.fn()
    }
  }
  const component = renderer.create(
    <Provider store={Store}>
      <BrowserRouter>
          <Switch>
            <LoginForm {...props}/>
          </Switch>
      </BrowserRouter>
    </Provider>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('it should handle submit when login button is clicked', ()=> {
  const preventDefault = jest.fn();
  const props = {
    auth: {
      isAuthenticated: true,
      user: {}
    },
    location: { 
      pathname: '/', 
      search: '', 
      hash: '', 
      state: undefined 
    },
    computedMatch: { 
      path: '/', 
      url: '/', 
      params: {}, 
      isExact: true 
    },
    history: {
      push: jest.fn()
    }
  }

  const wrapper = shallow(<LoginForm {...props}/>)
  wrapper.find('form').simulate('submit', { preventDefault });
  expect(preventDefault).toHaveBeenCalled();
});

test('it should handle change when input is added', ()=> {
  const target = { 
    name: 'email',
    value: 'ada'
  }

  const props = {
    auth: {
      isAuthenticated: true,
      user: {}
    },
    location: { 
      pathname: '/', 
      search: '', 
      hash: '', 
      state: undefined 
    },
    computedMatch: { 
      path: '/', 
      url: '/', 
      params: {}, 
      isExact: true 
    },
    history: {
      push: jest.fn()
    }
  }

  const wrapper = shallow(<LoginForm {...props}/>)
  wrapper.find('input').at(0).simulate('change', { target });
  expect(wrapper.state(target.name)).toEqual(target.value);
});