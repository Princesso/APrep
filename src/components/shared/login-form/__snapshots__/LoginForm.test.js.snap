// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form 1`] = `
<div
  className="form-holder"
>
  <form
    className="form"
    method="post"
    onSubmit={[Function]}
  >
    <h3>
      Login
    </h3>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      id="submitBtn"
      type="submit"
    >
      <span
        className="loading"
      >
         
        <p>
          Login
        </p>
      </span>
    </button>
    <p
      className="message"
    >
       
      <span />
       Have no account? 
      <span>
        <a
          className="linker"
          href="/register"
          onClick={[Function]}
        >
           Register
        </a>
      </span>
    </p>
  </form>
</div>
`;
